core/src/main/java/nu/yona/server/analysis/entities/Activity.java

```diff
@@ -0,0 +1,74 @@
+package nu.yona.server.analysis.entities;
+
+import java.util.Date;
+import java.util.UUID;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+import nu.yona.server.entities.EntityWithID;
+import nu.yona.server.entities.RepositoryProvider;
+import nu.yona.server.goals.entities.Goal;
+
+@Entity
+@Table(name = "ACTIVITIES")
+public class Activity extends EntityWithID
+{
+	public static ActivityRepository getRepository()
+	{
+		return (ActivityRepository) RepositoryProvider.getRepository(Activity.class, UUID.class);
+	}
+
+	private UUID userAnonymizedID;
+	private UUID goalID;
```

Comment:
In `GoalConflictMessage` these were IDs too. It would be OK to change them, but then we would have a problem with creating instances because `Goal` and `UserAnonymized` instances are not cached, but `GoalDTO` and `UserAnonymizedDTO` instances are. So if we change this from IDs to associations we need to retrieve the instances from the repository directly, isn't it?